Core/Resource/ResourceManager.cs

```diff
@@ -607,7 +607,7 @@ public uint CreateComputeShaderFromFile(string srcFile, string entryPoint, strin
             if (fileResource == null)
             {
                 fileResource = new ResourceFileHook(srcFile, new[] { newResource.Id });
-                ResourceFileWatcher._resourceFileHooks.Add(srcFile, fileResource);
+                ResourceFileWatcher._resourceFileHooks.TryAdd(srcFile, fileResource);
```

Comment:
I'm pretty sure it's just the fact that parallel access to this dictionary led to duplicate keys being added, since between the time it checked if the dictionary had the key and the time it actually added it if it didn't, another thread may have already added that value

on second thought, I am reverting this change - I'd hate for this to "work" but not WORK if we ever end up threading that stuff